src/wallets/router/router.types.ts

```diff
@@ -0,0 +1,28 @@
+import type React from "react";
+import type { RouteComponentProps as WouteComponentProps } from "wouter";
+import type { AuthRoutePath } from "~wallets/router/auth/auth.routes";
+import type { DashboardRoutePath } from "~wallets/router/dashboard/dashboard.routes";
+import type { PopupRoutePath } from "~wallets/router/popup/popup.routes";
+import type { WelcomeRoutePath } from "~wallets/router/welcome/welcome.routes";
+
+export interface CommonRouteProps<T = any>
+  extends Omit<WouteComponentProps, "params"> {
+  params: T;
+}
+
+export type BaseRoutePath = `/${string}`;
+
+export type RouteAuthType = "auth" | "anon";
+
+export interface RouteConfig<P extends BaseRoutePath = BaseRoutePath> {
+  key?: string;
+  path: P;
+  component: React.ComponentType<CommonRouteProps>;
+  authType?: RouteAuthType;
+}
+
+export type ArConnectRoutePath =
```

Comment:
Routes (across all apps) are now type-safe, which includes calls to `navigate()`, `<Link>` and `<Redirect>`.